erasure-utils.go

```diff
@@ -44,21 +44,22 @@ func newHash(algo string) hash.Hash {
 	}
 }
 
+// hashSum calculates the hash of the entire path and returns.
 func hashSum(disk StorageAPI, volume, path string, writer hash.Hash) ([]byte, error) {
 	startOffset := int64(0)
-	// Read until io.EOF.
+	// Read until io.EOF and write the read buffer to hash writer.
 	for {
-		buf := make([]byte, blockSizeV1)
-		n, err := disk.ReadFile(volume, path, startOffset, buf)
+		buf, err := disk.ReadFile(volume, path, startOffset, blockSizeV1)
```

Comment:
> This is incorrect io.ErrUnexpectedEOF has data in it.

it would have already done `writer.Write(buf)`
```
                if len(buf) > 0 {                                                      
                        writer.Write(buf)                                              
                        startOffset += int64(len(buf))                                 
                }                                                                      
                if err == io.EOF || err == io.ErrUnexpectedEOF {                       
                        break                                                          
                }           
```